src/Hedgehog/Gen.fs

```diff
@@ -458,20 +445,29 @@ module Gen =
 
     /// Generates a random globally unique identifier.
     let guid : Gen<System.Guid> =
-        gen { let! bs = byte |> array' 16 16
+        gen { let! bs = array' 16 16 <| bounded ()
```

Comment:
Nitpick: I could have used a type annotation here to make it clear we're asking for random bytes (?)